app/knights/knights.py

```diff
@@ -0,0 +1,45 @@
+from app.equipment.armour import Armour
+from app.equipment.potion import Potion
+from app.equipment.weapon import Weapon
+
+
+class Knight:
+    def __init__(self, name: str, power: int, hp: int) -> None:
+        self.name: str = name
+        self.power: int = power
+        self.hp: int = hp
+        self.armour: list[Armour] = []
+        self.weapon: Weapon | None = None
+        self.potion: Potion | None = None
+        self.protection = 0
+
+    def apply_armour(self, armour: list[dict]) -> None:
+        self.armour = Armour.set_of_armour(armour)
+
+    def apply_weapon(self, weapon: dict) -> None:
+        self.weapon = Weapon(weapon["name"], weapon["power"])
+
+    def apply_potion(self, potion: dict | None) -> None:
+        if potion is not None:
+            new_potion = Potion(potion["name"])
+            potion_effects = potion["effect"]
```

Comment:
Ensure that the `potion` dictionary contains the key `effect` and that it is a dictionary to avoid potential `KeyError` or `TypeError`. Consider adding validation or error handling for missing or incorrect keys.